typed-racket-lib/typed-racket/typecheck/tc-structs.rkt

```diff
@@ -64,6 +68,35 @@
     [(~or t:typed-struct t:typed-struct/exec)
      #'t.type-name]))
 
+(define-syntax-class expanded-props
+  #:literals (null list #%plain-app)
+  (pattern null
+           #:with (prop-names ...) #'()
+           #:with (prop-vals ...) #'())
+  (pattern (#%plain-app list (#%plain-app cons prop-names prop-vals) ...)))
+
+(define (tc/struct-prop-values form name)
+  (syntax-parse form
+    #:literals (define-values #%plain-app define-syntaxes begin #%expression let-values quote list cons make-struct-type values null)
+    [(define-values (struct-var r ...)
+       (let-values (((var1 r1 ...)
+                     (let-values ()
+                       (#%expression
+                        (let-values ()
+                          (#%plain-app make-struct-type _name _super-type _init-fcnt _auto-fcnt auto-v props:expanded-props r_args ...))))))
+         (#%plain-app values args-v ...)))
+     (for ([p (in-list (attribute props.prop-names))]
+           [pval (in-list (attribute props.prop-vals))])
+       (match (single-value p)
+         [(tc-result1: (StructProperty: ty))
+          (define sty (lookup-type-alias name parse-type))
+          (match-define (F: var) -Self)
+          (tc-expr/check pval (ret (subst var sty ty)))]
+         [(tc-result1: ty)
+          (tc-error "expected a struct type property but got ~a") ty]))]
```

Comment:
is returning `ty` in the error case here right? Should it be the weird `Error` type instead?

Reply:
This is a bug. I will fix it soon